provider/core/src/datagen/iter.rs

```diff
@@ -5,29 +5,26 @@
 //! Collection of iteration APIs for data providers.
 
```

Comment:
Praise: Thank you for doing these changes in a separate PR from the main crabbake PR